app/src/main/java/com/mionix/newsapp/repo/ProfileRepo.kt

```diff
@@ -0,0 +1,118 @@
+package com.mionix.newsapp.repo
+
+import android.content.ContentResolver
+import android.graphics.Bitmap
+import android.graphics.Matrix
+import android.net.Uri
+import android.provider.MediaStore
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.storage.FirebaseStorage
+import com.mionix.newsapp.model.DataProfile
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class ProfileRepo {
+    private var mAuth = FirebaseAuth.getInstance()
+    private var rootRef = FirebaseDatabase.getInstance().reference
+    suspend fun getDataProfile(): DataProfile = suspendCoroutine { data ->
+        val eventListener: ValueEventListener = object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                val name = mAuth.currentUser?.uid?.let {
+                    dataSnapshot.child("Users").child(it).child("name").getValue(String::class.java)
```

Comment:
Hardcode string kiểu này dễ sai lắm, mỗi ref link có thể bỏ vào 1 biến để tái sử dụng kiểu: 
val userNameRef = dataSnapshot.child("Users").child(it).child("name")